### PR TITLE
[Serializer] Allow to specify a single value in @Groups

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/19374#issuecomment-256688002
| License       | MIT
| Doc PR        | https://github.com/symfony/symfony-docs/pull/10314

Tiny DX improvement:

Before:

```php
use Symfony\Component\Serializer\Annotation\Groups;

class Product
{
    /**
     * @Groups({"admins"})
     */
    public $itemsSold;
}
```

Now allowed:

```php
use Symfony\Component\Serializer\Annotation\Groups;

class Product
{
    /**
     * @Groups("admins")
     */
    public $itemsSold;
}
```

### CODE REVIEW

**Comment by dunglas** [comment]
Travis error not related.

File: src/Symfony/Component/Serializer/Annotation/Groups.php [hunk 1]
```diff
@@ -39,23 +39,20 @@ public function __construct(array $data)
...
 
-        if (!is_array($data['value'])) {
-            throw new InvalidArgumentException(sprintf('Parameter of annotation "%s" must be an array of strings.', get_class($this)));
-        }
-
-        foreach ($data['value'] as $group) {
+        $value = (array) $data['value'];
+        foreach ($value as $group) {
             if (!is_string($group)) {
                 throw new InvalidArgumentException(sprintf('Parameter of annotation "%s" must be an array of strings.', get_class($this)));
```
Comment (GuilhemN) [inline src/Symfony/Component/Serializer/Annotation/Groups.php]: Shouldn't this message be adapted accordingly?

**Comment by ogizanagi** [comment]
👍 

Status: Reviewed

**Comment by xabbuh** [comment]
👍

**Comment by fabpot** [comment]
Thank you @dunglas.

**Pull request closed without merging**